### PR TITLE
feat: cleanup old flex mask patch, suppress Matmul bnb warn, and misc

### PR DESCRIPTION
**Pull request by NanoCode012**
# Description

- biggest change: flex mask patch removed as upstream is enough to account for packed seq
- suppress matmul bnb warnings
- validator check for cross entropy patches

## Summary by CodeRabbit

* **Documentation**
  * Added VRAM usage information to the training example.

* **New Features**
  * Added validation to prevent conflicting cross-entropy configuration options.

* **Bug Fixes**
  * Reduced warning noise from quantization library during training.

* **Chores**
  * Updated default training hyperparameters in example configuration.
  * Removed deprecated internal masking logic.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by NanoCode012** [comment]
Closes #3301 . Could not repro the linked issue problem. Cleanup from PR doesn't regress. Main doesn't show the issue too.

image

File: src/axolotl/utils/schemas/validation.py [hunk 1]
```diff
@@ -801,6 +794,36 @@ def check_xentropy_patch_conflicts(cls, data):
...
+        """Check for mutual exclusivity between cross entropy patch options.
+
+        Only one of the following can be enabled at a time:
+        - cut_cross_entropy (CutCrossEntropyPlugin)
+        - chunked_cross_entropy
+        - liger_cross_entropy (LigerPlugin)
+        - liger_fused_linear_cross_entropy (LigerPlugin)
+        """
+        ce_options = {
+            "cut_cross_entropy": data.get("cut_cross_entropy"),
```
Comment (winglian) [inline src/axolotl/utils/schemas/validation.py:809]: this one isn't required to set cce. simply adding the cce plugin enables cce

Comment (NanoCode012) [reply src/axolotl/utils/schemas/validation.py:809]: Yup, but adding the plugin sets this var to true, so we can check using this.

File: examples/olmo3/olmo3-7b-qlora.yaml [hunk 2]
```diff
@@ -42,10 +42,10 @@ wandb_watch:
 wandb_name:
 wandb_log_model:
 
-gradient_accumulation_steps: 4
+gradient_accumulation_steps: 2
 micro_batch_size: 2
 num_epochs: 1
-optimizer: adamw_bnb_8bit
+optimizer: adamw_8bit
```
Comment (salmanmohammadi) [inline examples/olmo3/olmo3-7b-qlora.yaml:48]: Does this need to ctrl+f replaced across the codebase? There's ~120 instances, mostly tests

Comment (NanoCode012) [reply examples/olmo3/olmo3-7b-qlora.yaml:48]: @SalmanMohammadi , is it preferred to use this over the other?
